tests/10apidoc/34room-messages.pl

```diff
@@ -151,7 +152,8 @@ sub matrix_send_room_text_message
             # With no params this does "forwards from END"; i.e. nothing useful
             params => {
                 dir => "b",
-                from => $token,
+                from => $next_batch,
```

Comment:
the spec could be clearer about this, but it's not intended that you can pass a `next_batch` token in `from`. Per https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-rooms-roomid-messages:

> `from`: ... this token can be obtained from a `prev_batch` token returned for each room by the sync API, or from a `start` or `end` token returned by a previous request to this endpoint.

I think that to do what this test wants, we need to do a *second* sync, and then paginate backwards from the `prev_batch` of the second sync. Which, incidentally, will also work around the synapse bug.